### DISCUSSION TITLE
What are the difference between cpptools.exe and cpptools-srv.exe ?

### DISCUSSION THREAD

**Discussion by stephane-archer** [microsoft/vscode-cpptools#8116] [Q&A] [answered]
Does vscode-cpptools is using Language Server?
From my understanding, the answer is yes.
I suppose:
`cpptools.exe` is the client/extension
`cpptools-srv.exe` is the server
But when I look at my process monitor I can see `cpptools.exe` doing a lot of work and I can't even see `cpptools-srv.exe` so I imagine I'm missing something.

**Comment by sean-mcmanus** [comment, marked as answer]
No -- cpptools.exe is the "server". The "client" is run inside the VS Code extension host as JavaScript. cpptools-srv.exe is a child process that runs IntelliSense processing. cpptools-srv.exe only runs if C_Cpp.intelliSenseEngine is set to "Default" and a file is opened.

**Reply by ericzhucode** [reply]
Are cpptools.exe and cpptools-srv.exe is kept as closed source? Besides, Is any document that guide us to use it via command line?

**Reply by bobbrow** [reply]
cpptools and cpptool-srv are closed source and cannot be used on the command line.  Our license requires that they be run in the context of VS Code or related Microsoft products.

**Reply by ericzhucode** [reply]
Thank you for your quick reply! Really clear and helpful
